atom/browser/ui/cocoa/atom_menu_controller.mm

```diff
@@ -58,6 +58,30 @@
     {@selector(clearRecentDocuments:), "clearrecentdocuments"},
 };
 
+// Called when adding a submenu to the menu and checks if the submenu, via its
+// |model|, has visible child items.
+bool MenuHasVisibleItems(const atom::AtomMenuModel* model) {
+  int count = model->GetItemCount();
+  for (int index = 0; index < count; index++) {
+    if (model->IsVisibleAt(index))
+      return true;
+  }
+  return false;
+}
+
+// Called when an empty submenu is created. This inserts a menu item labeled
+// "(empty)" into the submenu. Matches Windows behavior.
+NSMenu* MakeEmptySubmenu() {
+  base::scoped_nsobject<NSMenu> submenu([[NSMenu alloc] initWithTitle:@""]);
+  base::string16 empty_menu_title = base::ASCIIToUTF16("(empty)");
```

Comment:
In the past we've steered clear of hard-coded English strings.  Is there a translation in Chrome somewhere for this string we can leverage?

Reply:
@MarshallOfSound yes, native Windows/Linux apps use the same "empty" UX for submenus with no visible Menu Items!

And potentially, Chrome itself uses an int we can't leverage so i copied our approach to the Services submenu localization hardcode for this case

Reply:
Would be cool if we could use `IDS_APP_MENU_EMPTY_SUBMENU`, not sure if we get access to that in our build although we do use other `IDS_` strings in our code 🤔

Reply:
success!